src/corehost/cli/deps_format.cpp

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "deps_format.h"
+#include "utils.h"
+#include "trace.h"
+#include <unordered_set>
+#include <tuple>
+#include <array>
+#include <iterator>
+#include <cassert>
+#include <functional>
+
+const std::array<pal::char_t*, deps_entry_t::asset_types::count> deps_json_t::s_known_asset_types = {
+    _X("runtime"), _X("resources"), _X("native") };
+
+const deps_entry_t& deps_json_t::try_ni(const deps_entry_t& entry) const
+{
+    if (m_ni_entries.count(entry.asset_name))
+    {
+        assert(pal::strcmp(s_known_asset_types[0], _X("runtime")) == 0);
+        int index = m_ni_entries.at(entry.asset_name);
+        return m_deps_entries[0][index];
+    }
+    return entry;
+}
+
+bool deps_json_t::load_portable(const json_value& json, const pal::string_t& target_name, const rid_fallback_graph_t& rid_fallback_graph)
+{
+    portable_assets_t portable_assets;
+
+    for (const auto& package : json.at(_X("targets")).at(target_name).as_object())
+    {
+        const auto& targets = package.second.as_object();
+        auto iter = targets.find(_X("subtargets"));
```

Comment:
What is subtargets for? I dont see it in the spec (https://github.com/dotnet/cli/blob/rel/1.0.0/Documentation/specs/runtime-configuration-file.md).